### PR TITLE
feat: Add GitHub Action to auto assign PR author

### PR DESCRIPTION
**Pull request by mabdelhafiz94**
# Description

This GitHub action will auto-assign the PR author as an assignee and will not override the assignee if it's already assigned manually.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by jarvisswvl** [comment]
## Bitrise

[View test reports](https://addons-testing.bitrise.io/builds/fed11a76-30e1-420e-ba3e-a8b68395b746)

[View more details on Bitrise](https://app.bitrise.io/build/fed11a76-30e1-420e-ba3e-a8b68395b746)

| Build Artifacts |
|:-|
| [grype_output_vm_build.csv](https://app.bitrise.io/artifact/dfac9af1a30f46a1/download) |
| [grype_output_vm_build.json](https://app.bitrise.io/artifact/a4d7c7637fa3fae2/download) |

File: .github/workflows/assign-pr-author.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+name: Assign Author
+on: [pull_request]
+
+jobs:
+  label:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: technote-space/assign-author@v1.3.1
+      with:
+        repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (Tnation87) [inline .github/workflows/assign-pr-author.yml:12]: can we just add one empty line in the end?

Comment (mabdelhafiz94) [reply .github/workflows/assign-pr-author.yml:12]: I remember doing so, I think it was removed by GitHub file editor :thinking:

Comment (Tnation87) [reply .github/workflows/assign-pr-author.yml:12]: I think maybe you have the code reformatting option being on before committing? 👀

Comment (mabdelhafiz94) [reply .github/workflows/assign-pr-author.yml:12]: No it wasn't the IDE.. I created it using GitHub we interface

**Review by Tnation87 — APPROVED** [review]
Thank you 🙏 🚀

**Pull request merged**